gtwrap/pybind_wrapper.py

```diff
@@ -124,35 +131,29 @@ def _wrap_method(self, method, cpp_class, prefix, suffix, method_suffix=""):
                    suffix=suffix,
                ))
 
+        # Create __repr__ override
+        # We allow all arguments to .print() and let the compiler handle type mismatches.
         if method.name == 'print':
             # Redirect stdout - see pybind docs for why this is a good idea:
             # https://pybind11.readthedocs.io/en/stable/advanced/pycpp/utilities.html#capturing-standard-output-from-ostream
-            ret = ret.replace('self->', 'py::scoped_ostream_redirect output; self->')
+            ret = ret.replace('self->print', 'py::scoped_ostream_redirect output; self->print')
 
-            # __repr__() uses print's implementation:
-            type_list = method.args.to_cpp(self.use_boost)
-            if len(type_list) > 0 and type_list[0].strip() == 'string':
-                ret += '''{prefix}.def("__repr__",
-                    [](const {cpp_class} &a) {{
+            # Make __repr__() call print() internally
+            ret += '''{prefix}.def("__repr__",
+                    [](const {cpp_class}& self{opt_comma}{args_signature_with_names}){{
                         gtsam::RedirectCout redirect;
```

Comment:
then why do we still want this?

Reply:
https://github.com/borglab/wrap/pull/86

Reply:
Sorry, I mean if we use `py::scoped_ostream_redirect` then we don't need to use the GTSAM redirect anymore.

Reply:
We should tackle this in another PR so that we don't inadvertently break something. :confused: